hoomd/md/compute.py

```diff
@@ -0,0 +1,276 @@
+    those particles when requested. All specified :py:class:`ThermodynamicQuantities` objects can be added to a logger
+    for logging during a simulation, see :py:class:`hoomd.logging.Logger` for more details.
+
+    Examples::
+
+        f = filter.Type('A')
+        compute.ThermodynamicQuantities(filter=f)
+    """
+
+    def __init__(self, filter):
+        super().__init__(filter)
+
+    def _attach(self):
+        if isinstance(self._simulation.device, hoomd.device.CPU):
+            thermo_cls = _md.ComputeThermo
+        else:
+            thermo_cls = _md.ComputeThermoGPU
+        group = self._simulation.state.get_group(self._filter)
+        self._cpp_obj = thermo_cls(self._simulation.state._cpp_sys_def, group, "")
+        super()._attach()
+
+    @log
+    def kinetic_temperature(self):
+        R"""
+        :math:`kT`, instantaneous thermal energy of the group (in energy units).
+
+        Calculated as
+
+          .. math::
+
+            kT = 2 \cdot \frac{K}{N_{\mathrm{dof}}}
+        """
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.kinetic_temperature
+        else:
+            return None
+
+    @log
+    def pressure(self):
+        R"""
+        :math:`P`, instantaneous pressure of the group (in pressure units).
+
+        Calculated as
+
+          .. math::
+
+              W = \frac{1}{2} \sum_{i}\sum_{j \ne i} \vec{F}_{ij} \cdot \vec{r_{ij}} + \sum_{k} \vec{F}_{k} \cdot \vec{r_{k}}
+
+          where :math:`\vec{F}_{ij}` are pairwise forces between particles and :math:`\vec{F}_k` are forces due to explicit constraints, implicit rigid
+          body constraints, external walls, and fields. In 2D simulations,
+          :math:`P = (K + \frac{1}{2}\cdot W)/A` where :math:`A` is the area of the simulation box.
+          of the simulation box.
+        """
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.pressure
+        else:
+            return None
+
+    @log(flag='sequence')
+    def pressure_tensor(self):
+        R"""
+        (:math:`P_{xx}`, :math:`P_{xy}`, :math:`P_{xz}`, :math:`P_{yy}`, :math:`P_{yz}`, :math:`P_{zz}`).
+
+        Instantaneous pressure tensor of the group (in pressure units).
+
+          .. math::
+
+              P_{ij} = \left[  \sum_{k\in[0..N)} m_k v_{k,i} v_{k,j} +
+                               \sum_{k\in[0..N)} \sum_{l > k} \frac{1}{2} \left(\vec{r}_{kl,i} \vec{F}_{kl,j} + \vec{r}_{kl,j} \vec{F}_{kl, i} \right) \right]/V
+        """
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.pressure_tensor
+        else:
+            return None
+
+    @log
+    def kinetic_energy(self):
+        R""" :math:`K`, total kinetic energy of all particles in the group (in energy units). """
```

Comment:
Add a formula to make this computation explicit (K is the sum of K_trans and K_rot).